### ISSUE TITLE
Lint/UselessAccessModifier thinks a private attr_writer is useless

### ISSUE THREAD

**Issue by maxjacobson** [bbatsov/rubocop#2517]
Consider this code:

```ruby
class DogWalker
  attr_reader :dogs

  def initialize(*dogs)
    @dogs = dogs
  end

  def add_dog(dog)
    self.dogs = dogs + [dog]
  end

  private

  attr_writer :dogs
end

walker = DogWalker.new("milo", "lola")
walker.add_dog("Bean") # (uses the private attr_writer)
walker.dogs = [1, 2, 3] # (tries to use the attr_writer but fails because it's private)
```

The `attr_writer :dogs` there is private, but this generates a `Lint/UselessAccessModifier` offense even though sometimes it is useful to define a private setter (not sure how useful this particular example is :smile:)

**Comment by maxjacobson** [comment]
This is on the latest master (01953251719dfe2e07834097e27d6bae2adcc476) -- I'm trying it out on some of our code to see what's new :smile:

**Comment by alexdowad** [comment]
Good point. Just fixing this.

**Comment by alexdowad** [comment]
I had `attr`, `attr_reader`, and `attr_accessor` covered, but not `attr_writer`.

**Comment by alexdowad** [comment]
OK, just pushed a fix [here](https://github.com/bbatsov/rubocop/pull/2515).
